I note failures such as:
```
Check failure on line 103 in src/libraries/System.Text.Encoding.CodePages/src/System/Text/BaseCodePageEncoding.cs

Azure Pipelines
/ runtime-dev-innerloop (Build Linux x64 release SourceBuild)
src/libraries/System.Text.Encoding.CodePages/src/System/Text/BaseCodePageEncoding.cs#L103
src/libraries/System.Text.Encoding.CodePages/src/System/Text/BaseCodePageEncoding.cs(103,79): error CS0246: (NETCORE_ENGINEERING_TELEMETRY=Build) The type or namespace name 'Span<>' could not be found (are you missing a using directive or an assembly reference?)
```
However, I had built runtime on my own 64-bit system with this patches and got a clean bill:
```
Build succeeded.
    0 Warning(s)
    0 Error(s)
```
I am not sure why this is happening.

--- a/src/libraries/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryFormatterWriter.cs
+++ b/src/libraries/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryFormatterWriter.cs
@@ -268,7 +268,6 @@ namespace System.Runtime.Serialization.Formatters.Binary
                 if (!BitConverter.IsLittleEndian)
                 {
                     // we know that we are writing a primitive type, so just do a simple swap
-                    // Debug.Fail("Re-review this code if/when we start running on big endian systems");
                     for (int i = 0; i < bufferUsed; i += typeLength)
                     {
                         for (int j = 0; j < typeLength / 2; j++)

--- a/src/libraries/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryParser.cs
+++ b/src/libraries/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryParser.cs
@@ -894,7 +894,6 @@ namespace System.Runtime.Serialization.Formatters.Binary
                     if (!BitConverter.IsLittleEndian)
                     {
                         // we know that we are reading a primitive type, so just do a simple swap
-                        // Debug.Fail("Re-review this code if/when we start running on big endian systems");
                         for (int i = 0; i < bufferUsed; i += typeLength)
                         {
                             for (int j = 0; j < typeLength / 2; j++)

--- a/src/libraries/System.Text.Encoding.CodePages/src/System.Text.Encoding.CodePages.csproj
+++ b/src/libraries/System.Text.Encoding.CodePages/src/System.Text.Encoding.CodePages.csproj
@@ -84,4 +84,7 @@
     <Reference Include="System.Runtime.InteropServices" />
     <Reference Include="System.Threading" />
   </ItemGroup>
+  <ItemGroup Condition="!$(TargetFramework.StartsWith('$(NetCoreAppCurrent)'))">
+    <PackageReference Include="System.Memory" Version="$(SystemMemoryVersion)" />
+  </ItemGroup>
 </Project>

--- a/src/libraries/System.Text.Encoding.CodePages/src/System/Text/BaseCodePageEncoding.cs
+++ b/src/libraries/System.Text.Encoding.CodePages/src/System/Text/BaseCodePageEncoding.cs
@@ -100,9 +100,9 @@ namespace System.Text
             internal short unused1;             // Add an unused WORD so that CodePages is aligned with DWORD boundary.
         }
         private const int CODEPAGE_DATA_FILE_HEADER_SIZE = 44;
-        internal static unsafe void ReadCodePageDataFileHeader(Stream stream, Span<byte> codePageDataFileHeader)
+        internal static unsafe void ReadCodePageDataFileHeader(Stream stream, byte[] codePageDataFileHeader)
         {
-            stream.Read(codePageDataFileHeader);
+            stream.Read(codePageDataFileHeader, 0, codePageDataFileHeader.Length);
             if (!BitConverter.IsLittleEndian)
             {
                 fixed (byte* pBytes = &codePageDataFileHeader[0])
@@ -135,9 +135,9 @@ namespace System.Text
             [FieldOffset(0x24)]
             internal int Offset;            // DWORD
         }
-        internal static unsafe void ReadCodePageIndex(Stream stream, Span<byte> codePageIndex)
+        internal static unsafe void ReadCodePageIndex(Stream stream, byte[] codePageIndex)
         {
-            stream.Read(codePageIndex);
+            stream.Read(codePageIndex, 0, codePageIndex.Length);
             if (!BitConverter.IsLittleEndian)
             {
                 fixed (byte* pBytes = &codePageIndex[0])
@@ -178,9 +178,9 @@ namespace System.Text
             internal ushort ByteReplace;    // WORD     // default replacement bytes
         }
         private const int CODEPAGE_HEADER_SIZE = 48;
-        internal static unsafe void ReadCodePageHeader(Stream stream, Span<byte> codePageHeader)
+        internal static unsafe void ReadCodePageHeader(Stream stream, byte[] codePageHeader)
         {
-            stream.Read(codePageHeader);
+            stream.Read(codePageHeader, 0, codePageHeader!.Length);
             if (!BitConverter.IsLittleEndian)
             {
                 fixed (byte* pBytes = &codePageHeader[0])

--- a/src/libraries/System.Text.Encoding.CodePages/src/System/Text/BaseCodePageEncoding.netcoreapp.cs
+++ b/src/libraries/System.Text.Encoding.CodePages/src/System/Text/BaseCodePageEncoding.netcoreapp.cs
@@ -2,6 +2,7 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.IO;
+using System.Buffers.Binary;
 using System.Runtime.Serialization;
 using System.Runtime.CompilerServices;
 
@@ -9,6 +10,26 @@ namespace System.Text
 {
     internal abstract partial class BaseCodePageEncoding : EncodingNLS, ISerializable
     {
+        internal static unsafe void ReadCodePageIndex(Stream stream, Span<byte> codePageIndex)
+        {
+            stream.Read(codePageIndex);
+            if (!BitConverter.IsLittleEndian)
+            {
+                fixed (byte* pBytes = &codePageIndex[0])
+                {
+                    CodePageIndex* p = (CodePageIndex*)pBytes;
+                    char *pCodePageName = &p->CodePageName;
+                    for (int i = 0; i < 16; i++)
+                    {
+                            pCodePageName[i] = (char)BinaryPrimitives.ReverseEndianness((ushort)pCodePageName[i]);
+                    }
+                    p->CodePage = BinaryPrimitives.ReverseEndianness(p->CodePage);
+                    p->ByteCount = BinaryPrimitives.ReverseEndianness(p->ByteCount);
+                    p->Offset = BinaryPrimitives.ReverseEndianness(p->Offset);
+                }
+            }
+        }
+
         internal static unsafe EncodingInfo [] GetEncodings(CodePagesEncodingProvider provider)
         {
             lock (s_streamLock)

--- a/src/libraries/System.Text.Encoding.CodePages/src/System/Text/SBCSCodePageEncoding.cs
+++ b/src/libraries/System.Text.Encoding.CodePages/src/System/Text/SBCSCodePageEncoding.cs
@@ -33,15 +33,15 @@ namespace System.Text
         {
         }
 
-        internal static unsafe ushort ReadUInt16(byte *pByte)
+        internal static unsafe ushort ReadUInt16(byte* pByte)
         {
             if (BitConverter.IsLittleEndian)
             {
-              return *(ushort *)pByte;
+              return *(ushort*)pByte;
             }
             else
             {
-              return BinaryPrimitives.ReverseEndianness(*(ushort *)pByte);
+              return BinaryPrimitives.ReverseEndianness(*(ushort*)pByte);
             }
         }
 
@@ -106,7 +106,7 @@ namespace System.Text
                 {
                     for (int b = 0; b < 256; b++)
                     {
-                        char c = (char)ReadUInt16(pBuffer + 2*b);
+                        char c = (char)ReadUInt16(pBuffer + 2 * b);
                         // Don't want to force 0's to map Unicode wrong.  0 byte == 0 unicode already taken care of
                         if (c != 0 || b == 0)
                         {

